input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go

```diff
@@ -17,28 +17,30 @@
 
 package modeldecoderutil
 
-// Reslice increases the slice's capacity, if necessary, to match n.
-// If specified, the newFn function is used to create the elements
-// to populate the additional space appended to the slice.
-func Reslice[Slice ~[]model, model any](slice Slice, n int, newFn func() model) Slice {
-	if diff := n - cap(slice); diff > 0 {
-		// start of the extra space
-		idx := cap(slice)
+import (
+	"slices"
+)
 
-		// Grow the slice
-		// Note: append gives no guarantee on the capacity of the resulting slice
-		// and might overallocate as long as there's enough space for n elements.
-		slice = append([]model(slice)[:cap(slice)], make([]model, diff)...)
-		if newFn != nil {
-			// extend the slice to its capacity
-			slice = slice[:cap(slice)]
+// Reslice returns a slice with length n. It will reallocate the
+// slice if the capacity is not enough.
+func Reslice[Slice ~[]model, model any](slice Slice, n int) Slice {
+	if n-cap(slice) > 0 {
```

Comment:
nit: why `n-cap(slice)` instead of `n-len(slice)`, if its sole purpose is to avoid going into negative? I suppose the result is the same, but personally I prefer either `if n-len(slice) > 0` or `if n > cap(slice)` for clarity

Reply:
>why n-cap(slice) instead of n-len(slice), if its sole purpose is to avoid going into negative? 

If we have the capacity there is no point in even calling Grow.

Reply:
> but personally I prefer either if n-len(slice) > 0 or if n > cap(slice) for clarity

Not sure how it will be clearer honestly but I have no strong inclination so let me update with `n > cap(slice)`